### DISCUSSION TITLE
[Question]: Support for Widgets - WidgetKit, ActivityKit

### DISCUSSION THREAD

**Discussion by cmnstmntmn** [liveview-native/liveview-client-swiftui#1448] [Q&A]
### What happened?

@bcardarella Are there any plans for these technologies to be added to the list of add-ons?

https://developer.apple.com/widgets/

**Comment by bcardarella** [comment]
I'll let Carson weigh on this. Looking at the documentation these don't appear to be SwiftUI compatible views. WidgetKit looks to be part of `WidgetConfiguration` and ActivityKit appears to be mostly through UI outside of the app.

I do think we should look into how these would work with LVN but I don't yet know if these are something we could represent with LVN server side rendering.

**Comment by cmnstmntmn** [comment]
Great, ty @bcardarella!

**Comment by carson-katri** [comment]
WidgetKit/ActivityKit would probably not play well with LiveView. They can't be updated once they're rendered, so you wouldn't be able to use the socket to update the UI or anything like that.

The only thing I could see working, is if we allow rendering static SwiftUI markup on the server, then expose that as an endpoint like `/widget`. The client would fetch the markup and render it to static SwiftUI Views.
